Framework/HistogramData/test/HistogramIteratorTest.h

```diff
@@ -0,0 +1,412 @@
+                   });
+
+    // Assert
+    TSM_ASSERT("Frequencies did not match", result);
+  }
+
+  void test_iterate_over_histogram_frequencies_when_histogram_has_counts() {
+    // Arrange
+    Histogram hist(Points{1, 2, 3}, Counts{2, 3, 4});
+    Frequencies expectedFrequencies = hist.frequencies();
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedFrequencies.begin(),
+                   [](const HistogramItem &item, const double &frequency) {
+                     return item.frequency() == frequency;
+                   });
+
+    // Assert
+    TSM_ASSERT("Frequencies did not match", result);
+  }
+
+  void test_iterate_over_histogram_center_when_histogram_has_bins() {
+    // Arrange
+    Histogram hist(BinEdges{1, 2, 3, 4}, Counts{2, 3, 4});
+    Points expectedPoints = hist.points();
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedPoints.begin(),
+                   [](const HistogramItem &item, const double &point) {
+                     return item.center() == point;
+                   });
+
+    // Assert
+    TSM_ASSERT("Bin centers did not match", result);
+  }
+
+  void test_iterate_over_histogram_center_when_histogram_has_points() {
+    // Arrange
+    Histogram hist(Points{1, 2, 3}, Counts{2, 3, 4});
+    Points expectedPoints = hist.points();
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedPoints.begin(),
+                   [](const HistogramItem &item, const double &point) {
+                     return item.center() == point;
+                   });
+
+    // Assert
+    TSM_ASSERT("Bin centers did not match", result);
+  }
+
+  void test_iterate_over_histogram_width_when_histogram_has_bins() {
+    // Arrange
+    Histogram hist(BinEdges{1, 2, 3, 5}, Counts{2, 3, 4});
+    std::vector<double> expectedWidths = {1, 1, 2};
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedWidths.begin(),
+                   [](const HistogramItem &item, const double &width) {
+                     return item.width() == width;
+                   });
+
+    // Assert
+    TSM_ASSERT("Bin widths did not match", result);
+  }
+
+  void test_iterate_over_histogram_width_when_histogram_has_points() {
+    // Arrange
+    Histogram hist(Points{1, 3, 5}, Counts{2, 3, 4});
+    std::vector<double> expectedWidths = {2, 2, 2};
+
+    // Act
+    auto result =
+        std::equal(hist.begin(), hist.end(), expectedWidths.begin(),
+                   [](const HistogramItem &item, const double &width) {
+                     return item.width() == width;
+                   });
+
+    // Assert
+    TSM_ASSERT("Bin widths did not match", result);
+  }
+
+  void test_iterate_over_histogram_count_variances_when_histogram_has_counts() {
+    // Arrange
+    Histogram hist(BinEdges{1, 2, 3, 5}, Counts{2, 3, 4},
+                   CountVariances{3, 2, 1});
```

Comment:
Note that `Histogram` computes the uncertainties automatically if you do not specify them.